### PR TITLE
Overlay Refactor

### PR DESCRIPTION
**Pull request by mercury501**
Now draws each overlay with one DrawText call.

### CODE REVIEW

File: Wrappers/d3d8/Overlay.cpp [hunk 1]
```diff
@@ -155,118 +177,129 @@ void Overlay::DrawDebugOverlay(LPDIRECT3DDEVICE8 ProxyInterface)
...
+	{
+		DebugFont->DrawTextA(FontStruct.String, -1, &DropShadowRect, FontStruct.Format, TextColors.Black);
+		DebugFont->DrawTextA(FontStruct.String, -1, &FontStruct.Rect, FontStruct.Format, FontStruct.Color);
+	}
+}
 
-	TextStruct.String = OvlString.c_str();
-	TextStruct.Rect.top += TextSpacing;
-	TextStruct.Rect.bottom += TextSpacing;
+void Overlay::DrawMenuTestText(LPDIRECT3DDEVICE8 ProxyInterface, Overlay::D3D8TEXT FontStruct)
```
Comment (elishacloud) [inline Wrappers/d3d8/Overlay.cpp:242]: Can you rename this function `DrawMenuTestText()` to something more descriptive?  Maybe just remove the word "Test" from it?

**Comment by elishacloud** [comment]
Other than the two comments I added to the code, I found an issue with this code.  If I start the game at 1360x768 and then turn on both overlays and then switch to 800x600 then the info overlay disappears.

File: Wrappers/d3d8/Overlay.cpp (hunk 1 above)
Comment (mercury501) [reply Wrappers/d3d8/Overlay.cpp]: Of course!

**Comment by mercury501** [comment]
I renamed the function and added a check for changed resolution.
Thanks for spotting the bug!

**Comment by elishacloud** [comment]
I just realized you called this "MenuTest" all over the code. I wonder if it makes sense to call this "InfoMenu" or something like that instead?

**Comment by mercury501** [comment]
Reverted last name change!

**Comment by mercury501** [comment]
Should be good now!

**Comment by elishacloud** [comment]
Thanks @mercury501!

**Pull request merged**